subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -62,6 +66,47 @@ static inline bool dhcpv4_add_cookie(struct net_pkt *pkt)
 	return true;
 }
 
+#if defined(CONFIG_NET_DHCPV4_OPTION_CALLBACKS)
+static void *dhcpv4_option_callback_get_unique_types(uint8_t *types)
```

Comment:
The function no longer returns anything, so the return type should be `void`, not a pointer.